package/MDAnalysis/analysis/contacts.py

```diff
@@ -616,12 +609,17 @@ def run(self, store=True, force=False, start=0, stop=None, step=1, **kwargs):
             return None
 
         with openany(self.output, 'w') as out:
-            out.write("# q1 analysis\n# nref = {0:d}\n".format((self.nref)))
+            out.write("# q1 analysis\n# nref = %d\n" % (self.nref))
             out.write("# frame  q1  n1\n")
             records = []
             self.qavg *= 0  # average contact existence
             A, B = self.selections
-            for ts in self.universe.trajectory[start:stop:step]:
+            # determine the end_frame value to use:
+            total_frames = self.universe.trajectory.n_frames
+            if not end_frame:
```

Comment:
If step is a kwarg that defaults to None, just put it straight into the slice and let the Reader getitem handle it